sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -1140,15 +1143,26 @@ class CompactBlockProcessor internal constructor(
          * @return Latest block height wrapped in BlockHeight object, or null in case of failure
          */
         @VisibleForTesting
-        internal suspend fun fetchLatestBlockHeight(downloader: CompactBlockDownloader): BlockHeight? {
+        internal suspend fun fetchLatestBlockHeight(
+            downloader: CompactBlockDownloader,
+            sdkFlags: SdkFlags,
+        ): BlockHeight? {
             Twig.debug { "Fetching latest block height..." }
             val traceScope = TraceScope("CompactBlockProcessor.fetchLatestBlockHeight")
 
             var latestBlockHeight: BlockHeight? = null
 
             retryUpToAndContinue(FETCH_LATEST_BLOCK_HEIGHT_RETRIES) {
-                // TODO [#1772]: redirect to correct service mode after 2.1 release
-                when (val response = downloader.getLatestBlockHeight(ServiceMode.Direct)) {
+                when (
+                    val response =
+                        downloader.getLatestBlockHeight(
+                            if (sdkFlags.isTorEnabled) {
+                                ServiceMode.DefaultTor
+                            } else {
+                                ServiceMode.Direct
+                            }
+                        )
```

Comment:
All of these conditionals might be cleaner if `SdkFlags` had a method:
```kotlin
/**
 * Returns the `ServiceMode` to use for a particular `lightwalletd` query.
 *
 * - If Tor is enabled, this returns the provided `mode`.
 * - If Tor is disabled, this always returns `ServiceMode.Direct`.
 */
fun ifTor(mode: ServiceMode): ServiceMode {
    if (isTorEnabled) {
        mode
    } else {
        ServiceMode.Direct
    }
}
```
because in every case when `!sdkFlags.isTorEnabled` we want to use `ServiceMode.Direct`, so we only need to specify the Tor-enabled mode to use in each location. Then this would become:
```suggestion
                        downloader.getLatestBlockHeight(
                            sdkFlags.ifTor(ServiceMode.DefaultTor)
                        )
```

Reply:
This suggestion is mostly to avoid the line verbosity of Kotlin. The equivalent Swift code is pretty succinct, e.g.

```swift
mode: await sdkFlags.torEnabled ? .defaultTor : .direct
```

and equivalent Kotlin would remove most of my concern. But ensuring that `!sdkFlags.isTorEnabled` is handled consistently is also a useful benefit to this refactor.

Reply:
I was thinking the exact same thing actually

Reply:
done -> made it a little bit better using infix function